tycho-storage/src/postgres/entry_point.rs

```diff
@@ -0,0 +1,536 @@
+#![allow(unused)] //TODO: Remove this once we have usage in extractors
+use std::collections::HashMap;
+
+use diesel::prelude::*;
+use diesel_async::{AsyncPgConnection, RunQueryDsl};
+use tycho_common::{
+    models::{
+        blockchain::{EntryPoint, EntryPointTracingData, EntryPointWithData, TracedEntryPoint},
+        Chain,
+    },
+    storage::StorageError,
+    Bytes,
+};
+
+use super::{
+    orm::{
+        EntryPoint as ORMEntryPoint, NewEntryPoint, NewTracedEntryPoint, ProtocolComponent,
+        ProtocolSystem, TracedEntryPoint as ORMTracedEntryPoint,
+    },
+    schema::{self},
+    storage_error_from_diesel, PostgresGateway,
+};
+use crate::postgres::{
+    orm::{
+        EntryPointTracingType, NewEntryPointCallsAccount, NewEntryPointTracingData,
+        NewProtocolComponentHoldsEntryPoint,
+    },
+    PostgresError,
+};
+
+pub struct EntryPointFilter {
+    protocol_name: Option<String>,
```

Comment:
I'm assuming this is `ProtocolSystem`, not `ProtocolComponent` - please document